src/lib/autoinstall/widgets/ask_dialog.rb

```diff
@@ -0,0 +1,434 @@
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:notify]
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          if !@question.value.nil?
+            self.value = @question.value
+            return
+          end
+
+          if @question.default_value_script
+            default_from_script = run_script(@question.default_value_script)
+          end
+          self.value = default_from_script == "true" || @question.default == "true"
+        end
+      end
+
+      class InputField < CWM::InputField
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:hstretch, :notify, :notifyContextMenu]
+        end
+      end
+
+      class PasswordField < CWM::CustomWidget
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # Sets the password value
+        #
+        # This method updates both password widgets
+        #
+        # @return [String]
+        def value=(val)
+          new_value = val.to_s
+          Yast::UI.ChangeWidget(Id(widget_id), :Value, new_value)
+          Yast::UI.ChangeWidget(Id(confirmation_widget_id), :Value, new_value)
+        end
+
+        # Returns the password value
+        #
+        # @return [String]
+        def value
+          Yast::UI.QueryWidget(Id(widget_id), :Value)
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:hstretch, :notify, :notifyContextMenu]
+        end
+
+        def contents
+          VBox(
+            Password(Id(widget_id), Opt(*opt), label),
+            Password(Id(confirmation_widget_id), Opt(*opt), "")
+          )
+        end
+
+        # @macro seeCustomWidget
+        def validate
+          return true if value == confirmation_value
+
+          Yast::Popup.Message(
+            format(_("%{field}: the passwords do not match."), field: label)
+          )
+        end
+
+      private
+
+        def confirmation_widget_id
+          "#{widget_id}_confirmation"
+        end
+
+        def confirmation_value
+          Yast::UI.QueryWidget(Id(confirmation_widget_id), :Value)
+        end
+      end
+
+      class ComboBox < CWM::ComboBox
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          @question = question
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:notify, :immediate]
```

Comment:
why you need notify when you do not have handle?

Reply:
Because I need to stop the timer when the user changes the value.